### PR TITLE
[Subs 1027] handle post finish event

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/fileupload/errors/ErrorMessages.java [hunk 1]
```diff
@@ -5,7 +5,8 @@
  */
 public class ErrorMessages {
 
-    public static final String INVALID_PARAMETERS = "Invalid parameters";
-    public static final String NOT_SUPPORTED_EVENT = "Not supported event";
-    public static final String FILE_DOCUMENT_NOT_FOUND = "File document is not found in the database";
+    public static final String INVALID_PARAMETERS = "Invalid parameters.";
+    public static final String NOT_SUPPORTED_EVENT = "Not supported event.";
+    public static final String FILE_DOCUMENT_NOT_FOUND = "File document is not found in the database.";
+    public static final String FILE_IO_ERROR = "File could not be created.";
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/fileupload/errors/ErrorMessages.java]: Given the error text, should this variable be called FILE_CREATION_ERROR ?

File: src/main/java/uk/ac/ebi/subs/fileupload/eventhandlers/PostFinishEvent.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        return response;
+    }
+
+    ResponseEntity<Object> moveFile(File file, EventHandlerService eventHandlerService) {
+        ResponseEntity<Object> response;
+        try {
+            eventHandlerService.moveFile(file);
+            file.setStatus(FileStatus.READY_TO_ARCHIVE);
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/fileupload/eventhandlers/PostFinishEvent.java]: Is a file ready to archive at this point?

File: src/main/java/uk/ac/ebi/subs/fileupload/services/DefaultEventHandlerService.java [hunk 3]
```diff
@@ -55,4 +67,35 @@ public DefaultEventHandlerService(ValidationService validationService, FileRepos
...
+        String targetFilename = file.getFilename();
+
+        String fullTargetPath = assembleFullTargetPath(sourcePath, targetBasePath, targetPath, targetFilename);
+
+        String fullSourcePath = assembleFullSourcePath(sourcePath, sourceFileName);
+
+        Files.move(Paths.get(fullSourcePath), Paths.get(fullTargetPath), StandardCopyOption.ATOMIC_MOVE);
+    }
+
+    private String generateFolderName(String tusId) {
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/fileupload/services/DefaultEventHandlerService.java]: Do we need to generate a random number here? The submission ID is already random. You could take the first x chars, then the second x chars, and then the whole submission ID for folder names. e.g. if the submission ID is abcd-efgh-3490234234, the path could be /ab/cd/abcd-efgh-3490234234

File: src/main/java/uk/ac/ebi/subs/fileupload/services/EventHandlerService.java [hunk 4]
```diff
@@ -12,4 +16,18 @@
     ResponseEntity<Object> validateUploadRequest(TUSFileInfo tusFileInfo);
 
     ResponseEntity<Object> persistOrUpdateFileInformation(File file);
+
+    void moveFile(File file) throws IOException;
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/fileupload/services/EventHandlerService.java]: I'm not sure this needs to be in the shared service, looks like it is only needed in one place

File: src/main/java/uk/ac/ebi/subs/fileupload/util/FileStatus.java [hunk 5]
```diff
@@ -4,5 +4,5 @@
  * An enumerated list of file statuses.
  */
 public enum FileStatus {
-    INITIALIZED, UPLOADING, UPLOADED
+    INITIALIZED, UPLOADING, UPLOADED, READY_TO_ARCHIVE
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/fileupload/util/FileStatus.java]: As discussed, I think status names should describe the current status of the entity, READY_TO_ARCHIVE seems very forward looking. Please consider renaming this.

File: src/main/java/uk/ac/ebi/subs/fileupload/errors/ErrorMessages.java (hunk 1 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/fileupload/errors/ErrorMessages.java]: Agree

File: src/main/java/uk/ac/ebi/subs/fileupload/eventhandlers/PostFinishEvent.java (hunk 2 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/fileupload/eventhandlers/PostFinishEvent.java]: After team work: READY_TO_CHECK
Thanks :)

File: src/main/java/uk/ac/ebi/subs/fileupload/services/EventHandlerService.java (hunk 4 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/fileupload/services/EventHandlerService.java]: OK. I will have a look and refactor it

File: src/main/java/uk/ac/ebi/subs/fileupload/util/FileStatus.java (hunk 5 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/fileupload/util/FileStatus.java]: Done

**Pull request merged**